app/stac_api/models.py

```diff
@@ -214,19 +230,35 @@ def save(self, force_insert=False, force_update=False, using=None, update_fields
         # --> this could probably best be done with GeoDjango? (@Tobias)
         # I leave this open for the moment.
 
-        # check if collection's start_ and end_dates need to be updated
-        if self.collection.start_date is None:
-            self.collection.start_date = self.properties_datetime
-            self.collection.save()
+        try:
+
+            if self.collection.extent["temporal"]["interval"][0][0] is None:
 
-        elif self.properties_datetime < self.collection.start_date:
-            self.collection.start_date = self.properties_datetime
-            self.collection.save()
+                self.collection.extent["temporal"]["interval"][0][0] = self.properties_datetime
```

Comment:
`extent["temporal"]["interval"][0][0]` <- this smells a bit like it could cause troubles if for some reason the extent field doesn't have this exact structure. This should never happen, but still...

Reply:
Every access to the JSON-fields reps. dictionaries should now be inside a try ... except statement, so that at least an error message will be raised and logged, in case something unexpected happens